src/sidebar/directive/test/annotation-test.js

```diff
@@ -689,6 +690,45 @@ describe('annotation', function() {
       });
     });
 
+    describe('#flag()', function() {
+      beforeEach(function() {
+        fakeAnnotationMapper.flagAnnotation = sandbox.stub();
+      });
+
+      it(
+        'calls annotationMapper.flag() when an annotation is flagged',
+        function(done) {
+          var parts = createDirective();
+          fakeAnnotationMapper.flagAnnotation.returns($q.resolve());
+          parts.controller.flag();
+          assert.calledWith(fakeAnnotationMapper.flagAnnotation,
+              parts.annotation);
+          done();
+        }
+      );
+
+      it('flashes an error if the flag fails', function(done) {
+        var controller = createDirective().controller;
+        fakeAnnotationMapper.flagAnnotation.returns(Promise.reject({
+          status: 500,
+          statusText: 'Server error',
+        }));
+        controller.flag();
+        setTimeout(function () {
+          assert.calledWith(fakeFlash.error, '500 Server error', 'Flagging annotation failed');
+          done();
+        }, 0);
+      });
+
+      it('doesn\'t flash an error if the flag succeeds', function(done) {
+        var controller = createDirective().controller;
+        fakeAnnotationMapper.flagAnnotation.returns($q.resolve());
+        controller.flag();
+        assert.notCalled(fakeFlash.error);
```

Comment:
The test above uses a timeout to wait for the promise to resolve, as the callback for `Promise.reject()` doesn't get invoked until the next turn of the event loop. Can you check whether you need to do the same here. 🤔